### ISSUE TITLE
[a11y]: User avatar accessibility violations

### ISSUE THREAD

**Issue by anamikaanu96** [carbon-design-system/ibm-products#6353]
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

v2.53.0-rc.0

### Automated testing tool and ruleset

Manual testing

### Assistive technology

JAWS

### Description

- [ ] The label (Thomas J. Watson button) of the button is not descriptive and also the Visual Label is not same or matching with the Accessible 
        Label .

   Issue : 1. The given visual Label TW is not sufficient enough to understand the purpose of the  control.
               2. Provided Visual Label (TW ) , and existing  accessible Label  "Thomas  J Watson ) is not matching , Accessible Label not containing 
                   the words from the visual label in its beginning.

   Recommended Fix : Label expected to indicate the purpose  of the Control - Provide a  descriptive visual Label. For example - TW, Thomas J. 
   Watson profile button and also  make sure the visual label  is same as /matching with Accessible  label.

Note : Provide Descriptive Visual Label and use the same Text as Accessible Label by using ARIA Labelled by /ARIA Label Technics.

- [ ] There is no text alternative provided for the user Avatar image

   Recommended Fix : User Profile Image must be given with  appropriative  Text alternatives , to make this conveyed to screen reader users.
   For example :  Avatar of  Thomas J. Watson "

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/experimental-components-user-avatar-useravatar--with-image&globals=viewport:basic

### Steps to reproduce

Reported in all variants.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
